pkg/icingadb/db.go

```diff
@@ -166,7 +166,7 @@ func (db *DB) BuildInsertIgnoreStmt(into interface{}) (string, int) {
 	switch db.DriverName() {
 	case driver.MySQL:
 		// MySQL treats UPDATE id = id as a no-op.
-		clause = fmt.Sprintf(`ON DUPLICATE KEY UPDATE "%s" = "%s"`, columns[0], columns[0])
+		clause = `ON DUPLICATE KEY UPDATE "id" = "id"`
```

Comment:
What should I do then instead?

Reply:
From your commit message:

> The order of the columns varies each time when calling `db.BuildColumns()`, and it isn't guaranteed to be `id` the first one.

That is probably already the case without any of the changes in this PR, isn't it? Does anything break in this PR if you keep this code as is? If not, I'd keep this code and open an issue for it.

Reply:
Well, as Alex already commented there, mysql considers`ON DUPLICATE KEY UPDATE id = id` as no-op and when the used column is not an `id` then this could lead to unwanted updates.